### ISSUE TITLE
[bug] ensure data files page is included on sitemap for approved users with permission to submit data files

### ISSUE THREAD

**Issue by ADPennington** [raft-tech/TANF-app#2443]
## Description

@ttran-hub and I noticed that the data files link is missing from the sitemap during a11y testing for #2440. 

This issue seems to be tied to the deprecated Access Request property 

![image](https://user-images.githubusercontent.com/28515874/224145062-242d633d-907e-4dd0-8bb2-f47c18ec0d0a.png)

Older accounts that requested / were granted access via that deprecated flag _can_ access the Data Files page in the sitemap, whereas any newer users where that flag is set to false (see below) _cannot_ see the data files page. Presumably this issue stems from TDP still looking to the deprecated flag rather than the account approval status property as the source of truth.

![image](https://user-images.githubusercontent.com/28515874/224145632-e87d8856-2a04-462d-a6c5-6542cda00b38.png)

## What I expected to see

data files link on site map

## What I did see

![2440p2](https://user-images.githubusercontent.com/63075587/224093179-60874762-0002-400e-8243-1caf291f5e57.PNG)
![2440p1](https://user-images.githubusercontent.com/63075587/224093185-0975fa73-7502-4ca8-8c42-ab262d06be14.PNG)

## Other Helpful Information

+ URL of the page I was on:
+ Browser and version:
+ Operating System: ( Windows | MacOS X | Linux | Other )
+ Is the issue repeatable?: ( yes | no | don't know )
+ Has the issue occurred more than once?:

**Comment by reitermb** [comment]
Refined further with information detailing the presumed source of this issue 
cc @ADPennington @ttran-hub

**Comment by n0remac** [comment]
What I see in the code lines up with the previous research here.

The site map is still using the deprecated way of checking approval status. This function and all its uses should be removed: https://github.com/raft-tech/TANF-app/blob/develop/tdrs-frontend/src/utils/canViewAdmin.js#L4

And replaced with the new way of checking approval status with a selector as seen here:
https://github.com/raft-tech/TANF-app/blob/develop/tdrs-frontend/src/components/Home/Home.jsx#L39


The functions in canViewAdmin.js can all be replaces with selectors. The site map should use the PermissionsGuard made by @jtimpe:
https://github.com/raft-tech/TANF-app/blob/develop/tdrs-frontend/src/components/Header/Header.jsx#L115

This should be an easy fix.
